pkg/serviceaccount/externaljwt/plugin/testing/v1alpha1/externalsigner_mock.go

```diff
@@ -150,15 +150,18 @@ func (m *MockSigner) FetchKeys(ctx context.Context, req *v1alpha1.FetchKeysReque
 	keys := []*v1alpha1.Key{}
 
 	m.supportedKeysLock.RLock()
+	defer m.supportedKeysLock.RUnlock()
 	for id, k := range m.supportedKeys {
 		keys = append(keys, &v1alpha1.Key{
 			KeyId:                    id,
 			Key:                      k.Key,
 			ExcludeFromOidcDiscovery: k.ExcludeFromOidcDiscovery,
 		})
 	}
-	m.supportedKeysFetched.Broadcast()
-	m.supportedKeysLock.RUnlock()
+
+	// always defer it AFTER 'defer m.supportedKeysLock.RUnlock()'
+	// because, it must get executed BEFORE releasing the lock.
+	defer m.supportedKeysFetched.Broadcast()
```

Comment:
This doesn't really change anything I can see in the broadcast/unlock/return order

Reply:
You are right, I was operating with an incorrect understanding of defer coupled with return. 

I have updated the test case to add a 2s sleep after being notified of a key Fetch. That shall help remediate the flakes. Delay/sleep is required because we are relying on signals from external_signer to conclude that plugin is updated.

Reply:
will leave in the changes made to `externalsigner_mock.go` to enable releasing lock with defer